File: acgs2-core/sdk/go/policy_registry.go
package sdk

import (
	"bytes"
	"context"
	"encoding/json"
	"fmt"
	"net/http"
)

const (
	// ConstitutionalHash is the required hash for all ACGS-2 operations
	ConstitutionalHash = "cdd01ef066bc6cf2"
)

// PolicyRegistryService provides methods for managing policies through the Policy Registry API
type PolicyRegistryService struct {
	client *ACGS2Client
}

// NewPolicyRegistryService creates a new PolicyRegistryService instance
func NewPolicyRegistryService(client *ACGS2Client) *PolicyRegistryService {
	return &PolicyRegistryService{
		client: client,
	}
}

// ListPolicies retrieves a list of policies with optional filtering
func (s *PolicyRegistryService) ListPolicies(ctx context.Context, status *PolicyStatus, limit, offset int) ([]Policy, error) {
	url := fmt.Sprintf("%s/api/v1/policies", s.client.config.BaseURL)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	// Add query parameters
	q := req.URL.Query()
	if status != nil {
		q.Add("status", string(*status))
	}
	if limit > 0 {
		q.Add("limit", fmt.Sprintf("%d", limit))
	}
	if offset > 0 {
		q.Add("offset", fmt.Sprintf("%d", offset))
	}
	req.URL.RawQuery = q.Encode()

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var policies []Policy
	if err := json.NewDecoder(resp.Body).Decode(&policies); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return policies, nil
}

// CreatePolicy creates a new policy
func (s *PolicyRegistryService) CreatePolicy(ctx context.Context, req CreatePolicyRequest) (*Policy, error) {
	url := fmt.Sprintf("%s/api/v1/policies", s.client.config.BaseURL)

	requestData := map[string]interface{}{
		"name":              req.Name,
		"content":           map[string]interface{}{"rules": req.Rules},
		"format":            "json",
		"constitutionalHash": ConstitutionalHash,
	}

	if req.Description != nil {
		requestData["description"] = *req.Description
	}
	if len(req.Tags) > 0 {
		requestData["tags"] = req.Tags
	}
	if len(req.ComplianceTags) > 0 {
		requestData["complianceTags"] = req.ComplianceTags
	}

	jsonData, err := json.Marshal(requestData)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	httpReq, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}
	httpReq.Header.Set("Content-Type", "application/json")

	resp, err := s.client.doRequest(httpReq)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var policy Policy
	if err := json.NewDecoder(resp.Body).Decode(&policy); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &policy, nil
}

// GetPolicy retrieves a policy by ID
func (s *PolicyRegistryService) GetPolicy(ctx context.Context, policyID string) (*Policy, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s", s.client.config.BaseURL, policyID)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var policy Policy
	if err := json.NewDecoder(resp.Body).Decode(&policy); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &policy, nil
}

// UpdatePolicy updates an existing policy
func (s *PolicyRegistryService) UpdatePolicy(ctx context.Context, policyID string, req UpdatePolicyRequest) (*Policy, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s", s.client.config.BaseURL, policyID)

	updateData := make(map[string]interface{})
	if req.Name != nil {
		updateData["name"] = *req.Name
	}
	if req.Description != nil {
		updateData["description"] = *req.Description
	}
	if len(req.Rules) > 0 {
		updateData["rules"] = req.Rules
	}
	if req.Status != nil {
		updateData["status"] = string(*req.Status)
	}
	if len(req.Tags) > 0 {
		updateData["tags"] = req.Tags
	}
	if len(req.ComplianceTags) > 0 {
		updateData["complianceTags"] = req.ComplianceTags
	}
	updateData["constitutionalHash"] = ConstitutionalHash

	jsonData, err := json.Marshal(updateData)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	httpReq, err := http.NewRequestWithContext(ctx, "PATCH", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}
	httpReq.Header.Set("Content-Type", "application/json")

	resp, err := s.client.doRequest(httpReq)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var policy Policy
	if err := json.NewDecoder(resp.Body).Decode(&policy); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &policy, nil
}

// ActivatePolicy activates a policy
func (s *PolicyRegistryService) ActivatePolicy(ctx context.Context, policyID string) (*Policy, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s/activate", s.client.config.BaseURL, policyID)

	requestData := map[string]string{
		"constitutionalHash": ConstitutionalHash,
	}

	jsonData, err := json.Marshal(requestData)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	req, err := http.NewRequestWithContext(ctx, "PUT", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}
	req.Header.Set("Content-Type", "application/json")

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var policy Policy
	if err := json.NewDecoder(resp.Body).Decode(&policy); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &policy, nil
}

// VerifyPolicy verifies input against a policy
func (s *PolicyRegistryService) VerifyPolicy(ctx context.Context, policyID string, req PolicyVerificationRequest) (*PolicyVerificationResponse, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s/verify", s.client.config.BaseURL, policyID)

	jsonData, err := json.Marshal(req)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	httpReq, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}
	httpReq.Header.Set("Content-Type", "application/json")

	resp, err := s.client.doRequest(httpReq)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var result PolicyVerificationResponse
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &result, nil
}

// GetPolicyContent retrieves raw policy content
func (s *PolicyRegistryService) GetPolicyContent(ctx context.Context, policyID string) (interface{}, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s/content", s.client.config.BaseURL, policyID)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var content interface{}
	if err := json.NewDecoder(resp.Body).Decode(&content); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return content, nil
}

// GetPolicyVersions retrieves policy version history
func (s *PolicyRegistryService) GetPolicyVersions(ctx context.Context, policyID string) ([]PolicyVersion, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s/versions", s.client.config.BaseURL, policyID)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var versions []PolicyVersion
	if err := json.NewDecoder(resp.Body).Decode(&versions); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return versions, nil
}

// CreatePolicyVersion creates a new policy version
func (s *PolicyRegistryService) CreatePolicyVersion(ctx context.Context, policyID string, content interface{}, description *string) (*PolicyVersion, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s/versions", s.client.config.BaseURL, policyID)

	requestData := map[string]interface{}{
		"content":            content,
		"constitutionalHash": ConstitutionalHash,
	}

	if description != nil {
		requestData["description"] = *description
	}

	jsonData, err := json.Marshal(requestData)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	req, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}
	req.Header.Set("Content-Type", "application/json")

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var version PolicyVersion
	if err := json.NewDecoder(resp.Body).Decode(&version); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &version, nil
}

// GetPolicyVersion retrieves a specific policy version
func (s *PolicyRegistryService) GetPolicyVersion(ctx context.Context, policyID, version string) (*PolicyVersion, error) {
	url := fmt.Sprintf("%s/api/v1/policies/%s/versions/%s", s.client.config.BaseURL, policyID, version)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var policyVersion PolicyVersion
	if err := json.NewDecoder(resp.Body).Decode(&policyVersion); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &policyVersion, nil
}

// Authenticate performs user authentication
func (s *PolicyRegistryService) Authenticate(ctx context.Context, req AuthRequest) (*AuthResponse, error) {
	url := fmt.Sprintf("%s/api/v1/auth/token", s.client.config.BaseURL)

	jsonData, err := json.Marshal(req)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	httpReq, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}
	httpReq.Header.Set("Content-Type", "application/json")

	resp, err := s.client.doRequest(httpReq)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var authResp AuthResponse
	if err := json.NewDecoder(resp.Body).Decode(&authResp); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &authResp, nil
}

// ListBundles retrieves all policy bundles
func (s *PolicyRegistryService) ListBundles(ctx context.Context) ([]PolicyBundle, error) {
	url := fmt.Sprintf("%s/api/v1/bundles", s.client.config.BaseURL)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var bundles []PolicyBundle
	if err := json.NewDecoder(resp.Body).Decode(&bundles); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return bundles, nil
}

// CreateBundle creates a new policy bundle
func (s *PolicyRegistryService) CreateBundle(ctx context.Context, req CreateBundleRequest) (*PolicyBundle, error) {
	url := fmt.Sprintf("%s/api/v1/bundles", s.client.config.BaseURL)

	requestData := map[string]interface{}{
		"name":               req.Name,
		"policies":           req.Policies,
		"constitutionalHash": ConstitutionalHash,
	}

	if req.Description != nil {
		requestData["description"] = *req.Description
	}

	jsonData, err := json.Marshal(requestData)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal request: %w", err)
	}

	httpReq, err := http.NewRequestWithContext(ctx, "POST", url, bytes.NewBuffer(jsonData))
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}
	httpReq.Header.Set("Content-Type", "application/json")

	resp, err := s.client.doRequest(httpReq)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var bundle PolicyBundle
	if err := json.NewDecoder(resp.Body).Decode(&bundle); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &bundle, nil
}

// GetBundle retrieves a policy bundle by ID
func (s *PolicyRegistryService) GetBundle(ctx context.Context, bundleID string) (*PolicyBundle, error) {
	url := fmt.Sprintf("%s/api/v1/bundles/%s", s.client.config.BaseURL, bundleID)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var bundle PolicyBundle
	if err := json.NewDecoder(resp.Body).Decode(&bundle); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &bundle, nil
}

// GetActiveBundle retrieves the currently active policy bundle
func (s *PolicyRegistryService) GetActiveBundle(ctx context.Context) (*PolicyBundle, error) {
	url := fmt.Sprintf("%s/api/v1/bundles/active", s.client.config.BaseURL)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var bundle PolicyBundle
	if err := json.NewDecoder(resp.Body).Decode(&bundle); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return &bundle, nil
}

// HealthCheck performs a health check on the policy registry
func (s *PolicyRegistryService) HealthCheck(ctx context.Context) (map[string]interface{}, error) {
	url := fmt.Sprintf("%s/api/v1/health/policies", s.client.config.BaseURL)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var health map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&health); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return health, nil
}

// CacheHealth checks cache health
func (s *PolicyRegistryService) CacheHealth(ctx context.Context) (map[string]interface{}, error) {
	url := fmt.Sprintf("%s/api/v1/health/cache", s.client.config.BaseURL)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var health map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&health); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return health, nil
}

// ConnectionsHealth checks connections health
func (s *PolicyRegistryService) ConnectionsHealth(ctx context.Context) (map[string]interface{}, error) {
	url := fmt.Sprintf("%s/api/v1/health/connections", s.client.config.BaseURL)

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("failed to create request: %w", err)
	}

	resp, err := s.client.doRequest(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	var health map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&health); err != nil {
		return nil, fmt.Errorf("failed to decode response: %w", err)
	}

	return health, nil
}

File: src/core/sdk/python/acgs2_sdk/services/policy_registry.py
"""
ACGS-2 Policy Registry Service
Constitutional Hash: cdd01ef066bc6cf2
"""

from typing import TYPE_CHECKING, Any

try:
    from src.core.shared.types import JSONDict, JSONValue
except ImportError:
    JSONValue = str | int | float | bool | None | dict[str, Any] | list[Any]  # type: ignore[misc]
    JSONDict = dict[str, JSONValue]  # type: ignore[misc]

from acgs2_sdk.models import CreatePolicyRequest, Policy, PolicyStatus, UpdatePolicyRequest

if TYPE_CHECKING:
    from acgs2_sdk.client import ACGS2Client


class PolicyRegistryService:
    """Service for managing policies through the Policy Registry API."""

    def __init__(self, client: "ACGS2Client") -> None:
        """Initialize the policy registry service.

        Args:
            client: ACGS2 client instance
        """
        self._client = client
        self._base_path = "/api/v1"

    async def list_policies(
        self,
        status: PolicyStatus | None = None,
        limit: int = 50,
        offset: int = 0,
    ) -> list[Policy]:
        """List policies with optional filtering.

        Args:
            status: Filter by policy status
            limit: Maximum number of policies to return
            offset: Number of policies to skip

        Returns:
            List of policies
        """
        params: JSONDict = {"limit": limit, "offset": offset}
        if status:
            params["status"] = status.value

        data = await self._client.get(f"{self._base_path}/policies", params=params)
        return [Policy(**policy) for policy in data]

    async def create_policy(
        self,
        name: str,
        rules: list[JSONDict],
        description: str | None = None,
        format: str = "json",
        tags: list[str] | None = None,
        compliance_tags: list[str] | None = None,
    ) -> Policy:
        """Create a new policy.

        Args:
            name: Policy name
            rules: Policy rules
            description: Policy description
            format: Policy format (default: json)
            tags: Policy tags
            compliance_tags: Compliance tags

        Returns:
            Created policy
        """
        request = CreatePolicyRequest(
            name=name,
            rules=rules,
            description=description,
            tags=tags,
            compliance_tags=compliance_tags,
        )

        data = await self._client.post(
            f"{self._base_path}/policies",
            json={
                "name": request.name,
                "content": {"rules": request.rules},
                "format": format,
                "description": request.description,
            },
        )

        return Policy(**data)

    async def get_policy(self, policy_id: str) -> Policy:
        """Get a policy by ID.

        Args:
            policy_id: Policy ID

        Returns:
            Policy details
        """
        data = await self._client.get(f"{self._base_path}/policies/{policy_id}")
        return Policy(**data)

    async def update_policy(
        self,
        policy_id: str,
        name: str | None = None,
        description: str | None = None,
        rules: list[JSONDict] | None = None,
        status: PolicyStatus | None = None,
        tags: list[str] | None = None,
        compliance_tags: list[str] | None = None,
    ) -> Policy:
        """Update a policy.

        Args:
            policy_id: Policy ID
            name: New policy name
            description: New policy description
            rules: New policy rules
            status: New policy status
            tags: New policy tags
            compliance_tags: New compliance tags

        Returns:
            Updated policy
        """
        request = UpdatePolicyRequest(
            name=name,
            description=description,
            rules=rules,
            status=status,
            tags=tags,
            compliance_tags=compliance_tags,
        )

        update_data: JSONDict = {}
        if request.name is not None:
            update_data["name"] = request.name
        if request.description is not None:
            update_data["description"] = request.description
        if request.rules is not None:
            update_data["rules"] = request.rules
        if request.status is not None:
            update_data["status"] = request.status.value
        if request.tags is not None:
            update_data["tags"] = request.tags
        if request.compliance_tags is not None:
            update_data["compliance_tags"] = request.compliance_tags

        data = await self._client.patch(f"{self._base_path}/policies/{policy_id}", json=update_data)
        return Policy(**data)

    async def activate_policy(self, policy_id: str) -> Policy:
        """Activate a policy.

        Args:
            policy_id: Policy ID

        Returns:
            Activated policy
        """
        data = await self._client.put(f"{self._base_path}/policies/{policy_id}/activate")
        return Policy(**data)

    async def verify_policy(self, policy_id: str, input_data: JSONDict) -> JSONDict:
        """Verify input against a policy.

        Args:
            policy_id: Policy ID
            input_data: Input data to verify

        Returns:
            Verification result
        """
        data = await self._client.post(
            f"{self._base_path}/policies/{policy_id}/verify", json={"input": input_data}
        )
        return data

    async def get_policy_content(self, policy_id: str) -> JSONDict:
        """Get raw policy content.

        Args:
            policy_id: Policy ID

        Returns:
            Raw policy content
        """
        return await self._client.get(f"{self._base_path}/policies/{policy_id}/content")

    async def get_policy_versions(self, policy_id: str) -> list[JSONDict]:
        """Get policy version history.

        Args:
            policy_id: Policy ID

        Returns:
            List of policy versions
        """
        return await self._client.get(f"{self._base_path}/policies/{policy_id}/versions")

    async def create_policy_version(
        self,
        policy_id: str,
        rules: list[JSONDict],
        description: str | None = None,
    ) -> JSONDict:
        """Create a new policy version.

        Args:
            policy_id: Policy ID
            rules: New policy rules
            description: Version description

        Returns:
            Created policy version
        """
        data = await self._client.post(
            f"{self._base_path}/policies/{policy_id}/versions",
            json={
                "content": {"rules": rules},
                "description": description,
            },
        )
        return data

    async def get_policy_version(self, policy_id: str, version: str) -> JSONDict:
        """Get a specific policy version.

        Args:
            policy_id: Policy ID
            version: Version identifier

        Returns:
            Policy version details
        """
        return await self._client.get(f"{self._base_path}/policies/{policy_id}/versions/{version}")

    # Authentication endpoints
    async def authenticate(self, username: str, password: str) -> JSONDict:
        """Authenticate and get access token.

        Args:
            username: Username
            password: Password

        Returns:
            Authentication response with token
        """
        data = await self._client.post(
            f"{self._base_path}/auth/token", json={"username": username, "password": password}
        )
        return data

    # Bundle endpoints
    async def list_bundles(self) -> list[JSONDict]:
        """List all policy bundles.

        Returns:
            List of policy bundles
        """
        return await self._client.get(f"{self._base_path}/bundles")

    async def create_bundle(
        self,
        name: str,
        policies: list[str],
        description: str | None = None,
    ) -> JSONDict:
        """Create a policy bundle.

        Args:
            name: Bundle name
            policies: List of policy IDs
            description: Bundle description

        Returns:
            Created bundle
        """
        data = await self._client.post(
            f"{self._base_path}/bundles",
            json={
                "name": name,
                "policies": policies,
                "description": description,
            },
        )
        return data

    async def get_bundle(self, bundle_id: str) -> JSONDict:
        """Get a policy bundle by ID.

        Args:
            bundle_id: Bundle ID

        Returns:
            Bundle details
        """
        return await self._client.get(f"{self._base_path}/bundles/{bundle_id}")

    async def get_active_bundle(self) -> JSONDict:
        """Get the currently active policy bundle.

        Returns:
            Active bundle details
        """
        return await self._client.get(f"{self._base_path}/bundles/active")

    # Health check endpoints
    async def health_check(self) -> JSONDict:
        """Check policy registry health.

        Returns:
            Health check response
        """
        return await self._client.get(f"{self._base_path}/health/policies")

    async def cache_health(self) -> JSONDict:
        """Check cache health.

        Returns:
            Cache health response
        """
        return await self._client.get(f"{self._base_path}/health/cache")

    async def connections_health(self) -> JSONDict:
        """Check connections health.

        Returns:
            Connections health response
        """
        return await self._client.get(f"{self._base_path}/health/connections")
